homeassistant/components/shelly/sensor.py

```diff
@@ -24,14 +25,17 @@
     "current": [ELECTRICAL_CURRENT_AMPERE, sensor.DEVICE_CLASS_CURRENT],
     "deviceTemp": [None, sensor.DEVICE_CLASS_TEMPERATURE],
     "energy": [ENERGY_KILO_WATT_HOUR, sensor.DEVICE_CLASS_ENERGY],
+    "rollerEnergy": [ENERGY_KILO_WATT_HOUR, sensor.DEVICE_CLASS_ENERGY],
     "energyReturned": [ENERGY_KILO_WATT_HOUR, sensor.DEVICE_CLASS_ENERGY],
     "extTemp": [None, sensor.DEVICE_CLASS_TEMPERATURE],
     "humidity": [PERCENTAGE, sensor.DEVICE_CLASS_HUMIDITY],
     "luminosity": ["lx", sensor.DEVICE_CLASS_ILLUMINANCE],
     "overpowerValue": [POWER_WATT, sensor.DEVICE_CLASS_POWER],
     "power": [POWER_WATT, sensor.DEVICE_CLASS_POWER],
+    "rollerPower": [POWER_WATT, sensor.DEVICE_CLASS_POWER],
     "powerFactor": [PERCENTAGE, sensor.DEVICE_CLASS_POWER_FACTOR],
     "tilt": [DEGREE, None],
+    "totalWorkTime": [TIME_SECONDS, None],
```

Comment:
I don't think that we should expose everything as sensors. These are not settings, these are settings right ?

Reply:
Nope it's a status Shelly Air needs bulb replacement after X hours of work. This is why it makes sense to expose this.
So far is the only device with such sensor.

Simone

Reply:
We shouldn't have something that is relative time and changes on every update. Shouldn't we just make it a binary sensor saying something should be replaced ?

Reply:
What if we move to hours ?

Reply:
It's not allowed to store relative times in Home Assistant. So if we know the time it will stop working, we should add `replace_at` attribute and store a timestamp

Reply:
We cannot know the time it will stop as it depends on user usage. 

Same issue will raise for storing uptime of the device; this is really useful for monitoring and debugging. Is largely used by Shelly users.

Reply:
uptime is easy, you just store the timestamp the device booted. 

We are not allowed to store relative time in the state machine. For media players we do it like this:
 - total time used
 - total time used "updated at" timestamp

Then when the light is on, we can calculate total time used + now - "updated at".

Reply:
Brother integration doesn't follow the guideline: uptime is in days ("d") and there is a second attrib "str uptime".

I removed the totalworktime for now, will think about it later but that way we can have the other 2 sensors integrated.

Reply:
Finding a violation in another integration is no pass to violate it here. We need to think about the stability of the core, sending a state change every 5 seconds will slow it down and pollute the DB.

Reply:
I may expressed myself badly: I wanted to say that I was so unlucky to check Brother to understand how to do the job and I found a non-compliant integration. BTW, will raise a issue so can be fixed.

I love the approach you all have in looking at stability and performances and not only on new features.